### ISSUE TITLE
Exclamation mark before a single atom doesn't lead to the atom execution

### ISSUE THREAD

**Issue by vsbogd** [trueagi-io/hyperon-experimental#303]
The following unit test fails:
```rs
    #[test]
    fn metta_execute_non_expression() {
        let program = "!3";

        let metta = Metta::new(Shared::new(GroundingSpace::new()), Shared::new(Tokenizer::new()));
        metta.set_setting("type-check".into(), "auto".into());
        let result = metta.run(&mut SExprParser::new(program));
        assert_eq!(result, Ok(vec![vec![expr!({Number::Integer(3)})]]));
    }
```
But it passes when 3rd line is replaced by:
```rs
        let program = "! 3";
```
It is a confusing behavior, see https://github.com/trueagi-io/hyperon-experimental/pull/297#issuecomment-1554063066
The root cause is explained here https://github.com/trueagi-io/hyperon-experimental/pull/297#issuecomment-1554162372

**Comment by Necr0x0Der** [comment]
Well, I'm not precisely sure if it will not be confusing for someone that, say, `set!` or `set!me` are valid atom names, while `!set` is not. I have no strong opinion on the issue, but if we consider `!` as an atom by itself, then it should be less confusing that `!set` is one atom similar to `a-b` is also one atom instead of being 3 atoms. But if there are more people thinking that `!set` should be interpreted as two separate tokens, then I would not argue.

**Comment by vsbogd** [comment]
Ok, I agree, that parsing an exclamation mark as a separate atom in all situations could be confusing as well. I personally doesn't feel requiring a space in `! 3` is a big issue.
@ngeiswei, @tanksha, what do you think as active users?

**Comment by ngeiswei** [comment]
I am neutral on the manner, as long as it is well defined.  I do not have any problem with requiring a space.  If so, then I suppose the rule with `!`, as a standalone bang operator, is that it should be followed by a space or an open parenthesis, right?  What about multiple spaces, tabs or new lines?

**Comment by vsbogd** [comment]
Yes, it should be followed by any whitespace character or `(`.

**Comment by tanksha** [comment]
I agree that having `a ( or a white space` between the exclamation mark and the next expression makes coding simpler.
